flink-examples/src/main/java/io/pravega/examples/flink/primer/process/ExactlyOnceWriter.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+package io.pravega.examples.flink.primer.process;
+
+import io.pravega.client.stream.Stream;
+import io.pravega.connectors.flink.*;
+import io.pravega.connectors.flink.serialization.PravegaSerialization;
+import io.pravega.examples.flink.Utils;
+import io.pravega.examples.flink.primer.datatype.IntegerEvent;
+import io.pravega.examples.flink.primer.datatype.Constants;
+import io.pravega.examples.flink.primer.source.ThrottledIntegerEventProducer;
+import io.pravega.examples.flink.primer.util.FailingMapper;
+ import org.apache.flink.api.common.restartstrategy.RestartStrategies;
+ import org.apache.flink.api.common.time.Time;
+import org.apache.flink.api.java.utils.ParameterTool;
+import org.apache.flink.streaming.api.CheckpointingMode;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+
+/*
+ * Parameters
+ *     -scope  Pravega scope
+ *     -stream Pravega stream
+ *     -controller tcp://<controller_host>:<port>, e.g., tcp://localhost:9090
+ *     -num-events <number of events to be generated>
+ *
+ */
+public class ExactlyOnceWriter {
+    private static final long checkpointIntervalMillis = 100;
+    private static final Time txnTimeoutMillis = Time.milliseconds(30 * 1000);
+    private static final Time txnGracePeriodMillis = Time.milliseconds(30 * 1000);
+    private static final int defaultNumEvents = 50;
+
+    // read data from the time when the program starts
+    //private static final long readStartTimeMillis = Instant.now().toEpochMilli();
+
+    // Logger initialization
+    private static final Logger LOG = LoggerFactory.getLogger(ExactlyOnceWriter.class);
+
+    public static void main(String[] args) throws Exception {
+        LOG.info("Starting ExactlyOnce checker ...");
+
+        // initialize the parameter utility tool in order to retrieve input parameters
+        ParameterTool params = ParameterTool.fromArgs(args);
+
+        boolean exactlyOnce = Boolean.parseBoolean(params.get("exactlyonce", "true"));
+        int numEvents = Integer.parseInt(params.get("num-events", String.valueOf(defaultNumEvents)));
```

Comment:
I believe `ParameterTool` can support integer values too? Is there any reason why we are expecting a string format here?